pkg/controllers/management/auth/auth_config_test.go

```diff
@@ -120,7 +121,66 @@ func TestCleanupRuns(t *testing.T) {
 	}
 }
 
+func TestAuthConfigReset(t *testing.T) {
+	t.Parallel()
+
+	allFields := []string{"accessMode", "allowedPrincipalIds", "apiVersion", "kind", "metadata", "type"}
+	postResetFields := []string{"apiVersion", "kind", "metadata", "type"}
+
+	tests := []struct {
+		annotationValue string
+		retainedFields  []string
+	}{
+		{CleanupRancherLocked, allFields},
+		{CleanupUserLocked, allFields},
+		{CleanupUnlocked, postResetFields},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.annotationValue, func(t *testing.T) {
+			t.Parallel()
+			config := v3.AuthConfig{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Auth Config",
+					APIVersion: "management.cattle.io/v3",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        github.Name,
+					Annotations: map[string]string{CleanupAnnotation: test.annotationValue},
+				},
+				Type:                "githubConfig",
+				Enabled:             false,
+				AccessMode:          "unrestricted",
+				AllowedPrincipalIDs: []string{"user1", "user2"},
+			}
+
+			mockUsers := newMockUserLister()
+			controller := authConfigController{
+				cleanup:                 new(cleanupService),
+				authConfigsUnstructured: newMockAuthConfigClient(&config),
+				users:                   &mockUsers,
+			}
+
+			_, err := controller.sync("test", &config)
+			require.NoError(t, err)
+			u, err := controller.getUnstructured(&config)
+			require.NoError(t, err)
+
+			cfg := u.UnstructuredContent()
+			for _, field := range test.retainedFields {
```

Comment:
Nit: You aren't checking the length of the cfg's fields, so there could be other fields that you retained (which you didn't want to retain), which could allow your test to silently pass in cases where you would want a fail.